### DISCUSSION TITLE
RegisterPackage works for .NET 6, buts fails in .NET 4.5.1

### DISCUSSION THREAD

**Discussion by tpkurilla** [neolithos/neolua#168] [Q&A]
I'm calling some static C# functions from Lua. This code works in .NET 6, but fails in 4.5.1:

            var lua = new Neo.IronLua.Lua();
            LuaGlobal g = lua.CreateEnvironment();
            var localScope = new LuaTable();
            localScope["__metatable"] = new LuaTable();
            localScope.MetaTable["__index"] = g;
            g.RegisterPackage("alg", typeof(LuaAlg));
            var chunk1 = lua.CompileChunk("return 0", "return.lua", null);
            var result1 = g.DoChunk(chunk1);
            var chunk3 = lua.CompileChunk("return test", "ret-test", null);
            var result3 = chunk3.Run(localScope, null);
            var ch4 = lua.CompileChunk("lat, lon, hgt = alg.EFGtoLLH(0.0,0.0,0.0); return llh", "llh-test", null);
            var res4 = ch4.Run(localScope);

I thought it might be due to my targeting 4.7.2. So I added a 4.7.2 target to neolua, and get the same result.

Do the Framework DLLs have some specific thing I need to do to get it to work?

Again, same code (even calling the 4.7.2 library) works when using .NET 6 target, but fails for any Framework target.

Any suggestions?

**Comment by neolithos** [comment]
Is there any exception? The definition of `LuaAlg` is missing. May be there is something?

**Reply by tpkurilla** [reply]
My apologies for letting this drop.

There are no exceptions. LuaAlg is defined...the calls works perfectly in .NET 6. We were able to add the registration calls directly into Neo.Lua (causing some circular reference woes...but we handled it). Once we added the registration of the types into Neo.Lua all worked well.

I have recently returned to looking at this because the requirement has once again surfaced.

At this point we've converted a rather substantial code base from the v4.7.2 to .NET 8, and can now try to address the scripting issues.

Oddly, when I add 'net8.0' to the target list it fails to compile. One of the errors is:

"error CS1061: 'AppDomain' does not contain a definition for 'DefineDynamicAssembly' and no accessible extension method 'DefineDynamicAssembly' accepting a first argument of type 'AppDomain' could be found (are you missing a using directive or an assembly reference?)"

what's odd about that is the .NET documentation says it isn't even defined for .NET 6!

Do you plan supporting .NET 8 at some point...given that .NET 6 is end-of-life?

**Reply by neolithos** [reply]
I should fix this in the last version.

Do you use the latest?

**Reply by tpkurilla** [reply]
![neolua-error](https://github.com/user-attachments/assets/cbdc60a8-dc93-4be8-81d9-b10ace249cbe)

Just did a fresh clone. Build All succeeded. Added net8.0 to the target list, and this image shows the errors.

**Reply by neolithos** [reply]
Is changed in the last commit 77fa838

**Reply by tpkurilla** [reply]
Just re-synced, and it now works as expected.

Thanks!
